web/modules/custom/asset_distribution/asset_distribution.module

```diff
@@ -284,35 +284,44 @@ QUERY;
 }
 
 /**
- * Checks if the given distribution has a unique title within its release.
+ * Returns the parent of the distribution.
  *
- * A release might have different distributions associated with it. To avoid
- * confusion we should avoid having distributions with identical names within
- * the same release.
+ * If the distribution is already created, the parent will be retrieved from the
+ * database.
+ * If the distribution is new, it is created though the only allowed route and
+ * the parent can be retrieved from there.
+ * In any other case, skip the parent retrieval.
  *
  * @param \Drupal\rdf_entity\RdfInterface $asset_distribution
- *   The distribution to check.
+ *   The distribution entity.
  *
- * @return bool
- *   TRUE if the title is unique.
+ * @return \Drupal\rdf_entity\Entity\RdfInterface|null
+ *   The solution or release that this distribution belongs to.
  *
  * @throws \Exception
- *   Thrown when anything but an asset distribution entity is passed in.
+ *    Thrown when more than one parents are found.
  */
-function asset_distribution_title_is_unique(RdfInterface $asset_distribution) {
-  if ($asset_distribution->bundle() !== 'asset_distribution') {
-    throw new \Exception('Title uniqueness within a release can only be checked for asset distributions.');
+function asset_distribution_get_distribution_parent(RdfInterface $asset_distribution) {
+  if (!$asset_distribution->isNew()) {
+    $query = \Drupal::entityQuery('rdf_entity', 'OR')
+      ->condition('field_is_distribution', $asset_distribution->id())
+      ->condition('field_isr_distribution', $asset_distribution->id());
+    $parents = $query->execute();
+    if (count($parents) > 1) {
+      throw new Exception("More than one parents were found for distribution {$asset_distribution->label()}");
+    }
+    elseif (!empty($parents)) {
+      $rdf_entity_id = reset($parents);
+      return \Drupal::entityTypeManager()->getStorage('rdf_entity')->load($rdf_entity_id);
+    }
   }
-
-  $query = \Drupal::entityQuery('rdf_entity')
-    ->condition('label', $asset_distribution->label())
-    ->condition('rid', 'asset_distribution')
-    ->condition('og_audience', $asset_distribution->get('og_audience')->first()->getValue()['target_id']);
-  if (!empty($asset_distribution->id())) {
-    $query->condition('id', $asset_distribution->id(), '<>');
+  // If the distribution is now added, get the parent from the route.
+  elseif (\Drupal::routeMatch()->getRouteName() === 'asset_distribution.asset_release_asset_distribution.add') {
+    $parent = \Drupal::routeMatch()->getParameter('rdf_entity');
+    if ($parent instanceof RdfInterface && in_array($parent->bundle(), ['solution', 'asset_release'])) {
+      return $parent;
+    }
```

Comment:
This doesn't seem right, we shouldn't get the parent from the route parameters. It should be present on the entity even if it is new. If it is not present then we should implement an `#entity_builder` callback that populates the value.

After all not all distributions are created through the form, they might be federated, imported, or created during tests.